ballista/rust/scheduler/src/state/task_manager.rs

```diff
@@ -299,6 +320,23 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> TaskManager<T, U>
             }
         }
 
+        // spawn a delayed future to clean up job data on both Scheduler and Executors
+        let state = self.state.clone();
+        let job_id_str = job_id.to_owned();
+        let active_job_cache = self.active_job_cache.clone();
+        tokio::spawn(async move {
+            tokio::time::sleep(Duration::from_secs(CLEANUP_FINISHED_JOB_DELAY_SECS))
```

Comment:
(for interactive query finish in ms)  i think this will store a lot feature in heap. Just some opinion.